xbmc/input/InputCodingTable.h

```diff
@@ -30,6 +30,26 @@ class IInputCodingTable
   virtual int GetType() { return TYPE_WORD_LIST; }
 
   virtual ~IInputCodingTable() {}
+  /*! \brief Called for the active keyboard layout when it's loaded, stick any initialization here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+
+      \return true on success, false on any fatal errors
+  */
+  virtual bool Initialize() { return true; }
+
+  /*! \brief Called for the active keyboard layout when it's unloaded, stick any cleanup here
+
+      This won't be needed for most implementations so we don't set it =0 but provide a default
+      implementation.
+  */
+  virtual void UnInitialize() {}
+
+  /*! \brief Can be overridden if initialization is expensive to avoid calling initialize more than needed
+      \return true if initialization has beeen done and was successful, false otherwise.
+  */
+  virtual bool IsInitialized() const { return false; }
```

Comment:
This is inconsistent with the default implementation of `Initialize()` which always returns `true` but `IsInitialized()` always returns `false`. So calling `Initialize()` and then `IsInitialized()` will result in returning `true` but the check would then fail nonetheless.